docs/spl-docs/evaluation-functions/informational-functions.md

```diff
@@ -9,132 +9,252 @@ This function returns `TRUE` if the given value is Boolean.
+... | eval feedbackStatus=if(isnotnull(customerFeedback), "Received", "Awaiting")
+    | stats count by feedbackStatus
 ```
 
+**Explanation**:
+
+- The `eval` command uses `isnotnull` to check for non-`NULL` `customerFeedback` values, labeling these interactions as "Received" and others as "Awaiting".
+- The `stats` command then counts the number of interactions in each category, providing insights into the volume of feedback received versus pending.
+
 ## **typeof(&lt;value&gt;)**
 
-This function returns the data type of the given value.
+This function returns the data type of the given value, which can be a literal value or more commonly, a field name in your dataset.
 
 ### Usage
 
-You can use this function with the `eval` and `where` commands, within the `WHERE` clause of the `from` command, and as part of evaluation expressions with other commands.
+`typeof` is versatile and can be used with `eval` and `where` commands. It is particularly useful for understanding the data type of fields in your dataset, which can inform data processing and manipulation strategies.
+
+### Possible Outputs
+
+- **String**: Indicates the value is a string.
+- **Number**: Indicates the value is a numeric type.
+- **Boolean**: Indicates the value is a boolean (`TRUE` or `FALSE`).
+- **NULL**: Indicates the field is not present or its value is `NULL`.
 
 ### Example
 
-For raw values that are strings or numbers:
+1. To determine the data type of a field named `latency`:
 
 ```spl
-... | eval typeString=typeof("hello"), typeNumber=typeof(123)
+... | eval fieldType=typeof(latency)
 ```
 
-When passing a field name, which could be a string, number, or `NULL`:
+2. To handle different data types in processing using `if` statements:
 
 ```spl
-... | eval fieldType=typeof(latency)
+... | eval processData=if(typeof(field)=="String", "Process as string",
+                         if(typeof(field)=="Number", "Process as number",
+                         if(typeof(field)=="Boolean", "Process as boolean",
+                         if(typeof(field)=="Multivalue", "Process as multivalue",
+                         "Unknown type"))))
+```
+
+### Use-Case Example
+
+**Problem**: In a dataset containing various types of data, it's crucial to identify the data type of each field to apply appropriate processing techniques, especially when the data source is dynamic and the types of some fields can change.
+
+**Solution**: Use the `typeof` function to dynamically identify the data type of fields. This enables the application of data type-specific processing logic, ensuring data integrity and accuracy in analysis.
+
+**Implementation**:
+
+```spl
+... | eval fieldType=typeof(latency),
+         processData=case(
+           typeof(latency)="String", "Process as string",
+           typeof(latency)="Number", "Process as number",
+           typeof(latency)="Boolean", "Process as boolean",
+           typeof(latency)="NULL", "Process as null"
+         )
```

Comment:
You're just repeating your Example 2 (lines 232-237); you don't need this duplication. Either give a different example here or remove Example 2.

Either way, I think a little more detailed example here could be good. For example, maybe this field is supposed to be a number but if we get a string we'll try converting it (e.g., strip units, convert from hex, etc.), but booleans and NULLs are invalid. Or if you have a better idea, go with that. But the purpose of this is to make it easy for people to read an example and then be able to apply it to their own use case to solve their problem; by giving an example with steps after using `typeof`, we can make this easier for them